src/encoded/static/components/cart/cart.js

```diff
@@ -312,21 +543,61 @@ Facet.defaultProps = {
 /**
  * Display the file facets. These display the number of files involved -- not the number of
  * experiments with files matching a criteria. As the primary input to this component is currently
- * an array of experiment IDs while this facet displays all the files involved with those
+ * an array of experiment IDs while these facets displays all the files involved with those
  * experiments, this component begins by retrieving information about all relevant files from the
  * DB. Each time an experiment is removed from the cart while viewing the cart page, this component
  * again retrieves all relevant files for the remaining experiments.
  */
 class FileFacets extends React.Component {
+    /**
+     * Update the `facets` object by incrementing the count of the term within it selected by the
+     * `field` within the given `file`. This utility function only avoids duplicating code in the
+     * `assembleFacets` method.
+     * @param {object} facets Facet object to update - mutated!
+     * @param {string} field Field key within the facet to update
+     * @param {object} file File containing the term to add to the facet
+     */
+    static addFileTermToFacet(facets, field, file) {
+        const term = getObjectFieldValue(file, field);
+        if (term !== undefined) {
+            if (facets[field]) {
+                // The facet has been seen in this loop before, so add to or initialize
+                // the relevant term within this facet.
+                if (facets[field][term]) {
+                    facets[field][term] += 1;
+                } else {
+                    facets[field][term] = 1;
+                }
+            } else {
+                // The facet has not been seen in this loop before, so initialize it as
+                // well as the value of the relevant term within the facet.
+                facets[field] = {};
+                facets[field][term] = 1;
+            }
+        }
+    }
+
     constructor() {
         super();
         this.state = {
-            /** File search result facets to display */
-            displayedFacets: null,
             /** Tracks facet loading progress */
-            facetLoadProgress: 0,
+            facetLoadProgress: null,
         };
+
+        // Initialize the expanded state of every facet; only the first one is expanded by default.
+        const expandedStates = {};
+        displayedFacetFields.forEach((field, index) => {
+            expandedStates[field] = index === 0;
+        });
```

Comment:
I don't know if this is better, but the default `expandedStates` could probably be defined next to `displayedFacetFields`. Also it could just be defined explicitly since the first one is hard-coded to be the only expanded one. If we'll want to be able to redefine which facets are expanded, there could be a variable for the open ones that could be checked against here but I'm not sure if there's a good place to define that / if we'll want to be changing that.

Reply:
I’ll give that some thought and update this later. If I make `expandedStates` a module-level variable (I think that’s what you mean?) I’d have to clone it here since we have to give `this.state` its own objects each time. Here I’m generating a new object each time.

Reply:
I’d like to keep it this way. Right now if you change the order of `displayedFacetFields`, the first one gets opened by default with no other changes, and that already happened when Ben wanted the order changed earlier. If we decide to have specific ones opened by default, I’d prefer to make `displayedFacetFields` an array of objects that includes a boolean set to true for the opened-by-default ones, instead of a separate object that parallels `displayedFacetFields`.

Reply:
OK! Sounds good.